forms-web-app/tests/unit/validators/eligibility/decision-date.test.js

```diff
@@ -0,0 +1,140 @@
+      const rule = rules()[3].builder.build();
+
+      expect(rule.fields).toEqual(['decision-date']);
+      expect(rule.locations).toEqual(['body']);
+      expect(rule.optional).toBeFalsy();
+      // bit of a rubbish test
+      expect(rule.stack.length).toEqual(1);
+    });
+
+    it('should have an array of rules', () => {
+      expect(rules().length).toEqual(4);
+      expect(rules()[0].builder.fields[0]).toEqual('decision-date-day');
+      expect(rules()[1].builder.fields[0]).toEqual('decision-date-month');
+      expect(rules()[2].builder.fields[0]).toEqual('decision-date-year');
+      expect(rules()[3].builder.fields[0]).toEqual('decision-date');
+    });
+  });
+
+  describe('decisionDateExpiredMessage', () => {
+    it('should be defined', () => {
+      expect(decisionDateExpiredMessage).toEqual('Decision date expired');
+    });
+  });
+
+  describe('decisionDateCombiner', () => {
+    it('should combine the date from the given request object', () => {
+      expect(
+        decisionDateCombiner({
+          body: {
+            'decision-date-day': 11,
+            'decision-date-month': 3,
+            'decision-date-year': 2020,
+          },
+        })
+      ).toEqual('2020-3-11');
+    });
+  });
+
+  describe('combinedDecisionDateFieldValidator', () => {
+    it('should throw if given an invalid date', () => {
+      expect(() => combinedDecisionDateFieldValidator({})).toThrowError(
+        new Error(JSON.stringify({ msg: 'Invalid date' }))
+      );
+    });
+
+    it('should throw if given an invalid date - unusual format', () => {
+      const req = {
+        body: {
+          'decision-date-day': 1111,
+          'decision-date-month': 3333,
+          'decision-date-year': 2020909,
+        },
+      };
+      expect(() => combinedDecisionDateFieldValidator(req)).toThrowError(
+        new Error(JSON.stringify({ msg: 'Invalid date' }))
+      );
+    });
+
+    it('should throw if decision date is too far in the future', () => {
+      const req = {
+        body: {
+          'decision-date-day': 11,
+          'decision-date-month': 6,
+          'decision-date-year': 2130,
+        },
+      };
+      expect(() => combinedDecisionDateFieldValidator(req)).toThrowError(
+        new Error(JSON.stringify({ msg: 'Invalid date' }))
+      );
+    });
+
+    xit('should throw if decision date is too far in the past', () => {
+      const req = {
+        body: {
+          'decision-date-day': 5,
+          'decision-date-month': 6,
+          'decision-date-year': 1920,
+        },
+      };
+      expect(() => combinedDecisionDateFieldValidator(req)).toThrowError(
+        new Error(
+          JSON.stringify({ msg: 'Decision date expired', deadlineDate: '1920-08-28T22:59:59.999Z' })
+        )
+      );
+    });
+
+    it('should resolve if valid', async () => {
+      const now = new Date();
+      const req = {
+        body: {
+          'decision-date-day': now.getDay(),
```

Comment:
@a6software I think this should be `getDate()` - `getDay` gives the day number and is failing on Sunday (day `0`)

Reply:
Fixed, thanks